File: src/main/java/ru/ifmo/is/mfl/common/search/SearchCriteria.java
package ru.ifmo.is.mfl.common.search;

import lombok.*;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class SearchCriteria {

  private String filterKey;
  private String operation;
  private Object value;
  private String dataOption;

  public SearchCriteria(String filterKey, String operation, Object value) {
    this.filterKey = filterKey;
    this.operation = operation;
    this.value = value;
  }
}

File: client/src/interfaces/search/dto/SearchCriteria.ts
export type Operation = 'cn' | 'nc' | 'eq' | 'ne' | 'be'
               | 'af' | 'bw' | 'bn' | 'sc' | 'sd'
               | 'se' | 'sn' | 'ew' | 'en' | 'nu'
               | 'nn' | 'gt' | 'ge' | 'lt' | 'le';

export interface SearchCriteria {
    filterKey: string;
    operation: Operation;
    value: any;
}